From 13e646460432a8daf48271d4e3b1b19780e642be Mon Sep 17 00:00:00 2001
From: Nikolay Kulikov <nikolayof23@gmail.com>
Date: Thu, 9 Jul 2026 21:15:08 +0300
Subject: [PATCH 03808/15333] staging: rtl8723bs: remove dead fields from
 struct hal_com_data

These fields exist but are never used, and nothing is ever written to
them, so remove them.

Signed-off-by: Nikolay Kulikov <nikolayof23@gmail.com>
Link: https://patch.msgid.link/20260709182627.16327-2-nikolayof23@gmail.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723bs/include/hal_data.h | 69 --------------------
 1 file changed, 69 deletions(-)

diff --git a/drivers/staging/rtl8723bs/include/hal_data.h b/drivers/staging/rtl8723bs/include/hal_data.h
index a03e32e4884c..2c630c6b5f97 100644
--- a/drivers/staging/rtl8723bs/include/hal_data.h
+++ b/drivers/staging/rtl8723bs/include/hal_data.h
@@ -188,25 +188,14 @@ struct hal_com_data {
 	u8 rf_chip;
 	u8 PackageType;
 
-	u8 InterfaceSel;
-	u8 framesync;
-	u32 framesyncC34;
-	u8 framesyncMonitor;
-	u8 DefaultInitialGain[4];
 	/*  EEPROM setting. */
-	u16 EEPROMVID;
-	u16 EEPROMSVID;
-
 	u8 EEPROMCustomerID;
-	u8 EEPROMSubCustomerID;
 	u8 EEPROMVersion;
 	u8 EEPROMRegulatory;
 	u8 EEPROMThermalMeter;
 	u8 EEPROMBluetoothCoexist;
 	u8 EEPROMBluetoothType;
 	u8 EEPROMBluetoothAntNum;
-	u8 EEPROMBluetoothAntIsolation;
-	u8 EEPROMBluetoothRadioShared;
 	u8 bTXPowerDataReadFromEEPORM;
 	u8 bDisableSWChannelPlan; /*  flag of disable software change channel plan */
 
@@ -226,24 +215,12 @@ struct hal_com_data {
 
 	u8 Regulation2_4G;
 
-	u8 TxPwrInPercentage;
-
-	u8 TxPwrCalibrateRate;
 	/*  TX power by rate table */
 	/*  RF: at most 2 = AB = 0/1 */
 	/*  CCK = 0 OFDM = 1 HT-MCS 0-7 = 2 */
-	u8 TxPwrByRateTable;
-	u8 TxPwrByRateBand;
 	s8 TxPwrByRateOffset[MAX_RF_PATH_NUM][TX_PWR_BY_RATE_NUM_RATE];
 	/*  */
 
-	/* 2 Power Limit Table */
-	u8 TxPwrLevelCck[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];
-	u8 TxPwrLevelHT40_1S[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];	/*  For HT 40MHZ pwr */
-	u8 TxPwrLevelHT40_2S[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];	/*  For HT 40MHZ pwr */
-	s8	TxPwrHt20Diff[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];/*  HT 20<->40 Pwr diff */
-	u8 TxPwrLegacyHtDiff[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];/*  For HT<->legacy pwr diff */
-
 	/*  Power Limit Table for 2.4G */
 	s8	TxPwrLimit_2_4G[MAX_REGULATION_NUM]
 						[MAX_2_4G_BANDWIDTH_NUM]
@@ -254,52 +231,26 @@ struct hal_com_data {
 	/*  Store the original power by rate value of the base of each rate section of rf path A & B */
 	u8 TxPwrByRateBase2_4G[MAX_RF_PATH_NUM][MAX_RATE_SECTION_NUM];
 
-	/*  For power group */
-	u8 PwrGroupHT20[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];
-	u8 PwrGroupHT40[RF_PATH_MAX_92C_88E][CHANNEL_MAX_NUMBER];
-
-
-
-
-	u8 PGMaxGroup;
-	u8 LegacyHTTxPowerDiff;/*  Legacy to HT rate power diff */
-	/*  The current Tx Power Level */
-	u8 CurrentCckTxPwrIdx;
-	u8 CurrentOfdm24GTxPwrIdx;
-	u8 CurrentBW2024GTxPwrIdx;
-	u8 CurrentBW4024GTxPwrIdx;
-
 	/*  Read/write are allow for following hardware information variables */
 	u8 pwrGroupCnt;
 	u32 MCSTxPowerLevelOriginalOffset[MAX_PG_GROUP][16];
-	u32 CCKTxPowerLevelOriginalOffset;
 
 	u8 CrystalCap;
-	u32 AntennaTxPath;					/*  Antenna path Tx */
-	u32 AntennaRxPath;					/*  Antenna path Rx */
 
-	u8 PAType_2G;
-	u8 LNAType_2G;
-	u8 ExternalPA_2G;
-	u8 ExternalLNA_2G;
 	u8 TypeGLNA;
 	u8 TypeGPA;
 	u8 TypeALNA;
 	u8 TypeAPA;
 	u8 RFEType;
 	u8 BoardType;
-	u8 ExternalPA;
 	u8 bIQKInitialized;
 	bool		bLCKInProgress;
 
 	bool		bSwChnl;
 	bool		bSetChnlBW;
-	bool		bChnlBWInitialized;
 	bool		bNeedIQK;
 
-	u8 bLedOpenDrain; /*  Support Open-drain arrangement for controlling the LED. Added by Roger, 2009.10.16. */
 	u8 TxPowerTrackControl; /* for mp mode, turn off txpwrtracking as default */
-	u8 b1x1RecvCombine;	/*  for 1T1R receive combining */
 
 	u32 AcParam_BE; /* Original parameter for BE, use for EDCA turbo. */
 
@@ -307,9 +258,6 @@ struct hal_com_data {
 
 	u32 RfRegChnlVal[2];
 
-	/* RDG enable */
-	bool	 bRDGEnable;
-
 	/* for host message to fw */
 	u8 LastHMEBoxNum;
 
@@ -320,46 +268,30 @@ struct hal_com_data {
 	u8 RegFwHwTxQCtrl;
 	u8 RegReg542;
 	u8 RegCR_1;
-	u8 Reg837;
-	u8 RegRFPathS1;
 	u16 RegRRSR;
 
 	u8 CurAntenna;
 	u8 AntDivCfg;
 	u8 AntDetection;
-	u8 TRxAntDivType;
 	u8 ant_path; /* for 8723B s0/s1 selection */
 
 	u8 u1ForcedIgiLb;			/*  forced IGI lower bound */
 
 	u8 bDumpRxPkt;/* for debug */
 	u8 bDumpTxPkt;/* for debug */
-	u8 FwRsvdPageStartOffset; /* 2010.06.23. Added by tynli. Reserve page start offset except beacon in TxQ. */
 
 	/*  2010/08/09 MH Add CU power down mode. */
 	bool		pwrdown;
 
-	/*  Add for dual MAC  0--Mac0 1--Mac1 */
-	u32 interfaceIndex;
-
 	u8 OutEpQueueSel;
 	u8 OutEpNumber;
 
 	/*  2010/12/10 MH Add for USB aggregation mode dynamic scheme. */
 	bool		UsbRxHighSpeedMode;
 
-	/*  2010/11/22 MH Add for slim combo debug mode selective. */
-	/*  This is used for fix the drawback of CU TSMC-A/UMC-A cut. HW auto suspend ability. Close BT clock. */
-	bool		SlimComboDbg;
-
-	/* u8 AMPDUDensity; */
-
 	/*  Auto FSM to Turn On, include clock, isolation, power control for MAC only */
 	u8 bMacPwrCtrlOn;
 
-	u8 RegIQKFWOffload;
-	struct submit_ctx	iqk_sctx;
-
 	enum rt_ampdu_burst	AMPDUBurstMode; /* 92C maybe not use, but for compile successfully */
 
 	u32 		sdio_himr;
@@ -390,6 +322,5 @@ struct hal_com_data {
 };
 
 #define GET_HAL_DATA(__padapter)	((struct hal_com_data *)((__padapter)->HalData))
-#define RT_GetInterfaceSelection(_Adapter)	(GET_HAL_DATA(_Adapter)->InterfaceSel)
 
 #endif /* __HAL_DATA_H__ */
-- 
2.55.0

